examples/manila-csi-plugin/nfs/topology-aware/storageclass.yaml

```diff
@@ -0,0 +1,37 @@
+# Topology constraints example:
+#
+# Let's have two Manila AZs: manila-zone-1, manila-zone-2
+# Let's have six Nova AZs: nova-{1..6}
+#
+# manila-zone-1 is accessible from nodes in nova-{1,2,3} only
+# manila-zone-2 is accessible from nodes in nova-{4,5,6} only
+#
+# We're provisioning into manila-zone-1
+# allowedTopologies reflects the topology constraints
+
+apiVersion: storage.k8s.io/v1
+kind: StorageClass
+metadata:
+  name: csi-manila-nfs-az
+provisioner: nfs.manila.csi.openstack.org
+parameters:
+  type: default
+  availability: manila-zone-1
+  
+  csi.storage.k8s.io/provisioner-secret-name: csi-manila-secrets
+  csi.storage.k8s.io/provisioner-secret-namespace: default
+  csi.storage.k8s.io/node-stage-secret-name: csi-manila-secrets
+  csi.storage.k8s.io/node-stage-secret-namespace: default
+  csi.storage.k8s.io/node-publish-secret-name: csi-manila-secrets
+  csi.storage.k8s.io/node-publish-secret-namespace: default
+allowedTopologies:
+  - matchLabelExpressions:
+    - key: topology.manila.csi.openstack.org/zone
+      values:
+        - nova-1
+        - nova-2
+        - nova-3
+    # ...or you may use other node labels:
```

Comment:
As I've said in the other comments, due to this loose coupling between Manila and compute AZs, the selected topology is not used during the actual storage provisioning. So there's not much point in using `WaitForFirstConsumer` binding mode...unless I'm missing something?

Reply:
@gman0 @gouthampacha sorry about the questions, trying to understand lil better here..
'availability' is manila AZ where volume is provisioned
Allowed Topologies , say what all node AZ accessible with manila AZ provided, Am I right?
So when pod is scheduled , you would want to still take into account, it should be scheduled where manila AZ is accessible right , that is, it should scheduled on nodes where node AZ , nova1, nova2 or nova3 right?, so that will be taken care only when u specify delayed binding mode AFAIK . please correct if Im wrong

Reply:
@ramineni - not so, because _we_ have no programmatic way to tell if manila AZ (ex: "manila-zone-1") is actually reachable by any of the nova AZs (ex: "nova1", "nova2", "nova3"). This reachability will only be known to the deployer/operator. 

So, both delayed binding or immediate binding will have the same effect - the onus is on the deployer to set these selectors correctly, because both AZs are something they conceive (or know about through other means).

Reply:
With manila, volume reachability is more reflective of network topology than compute  topology, compute failure domains.  It doesn't really map to neutron AZs [1] directly either because these reflect availability of neutron agents like DHCP, L3 agent, etc.

[1] https://docs.openstack.org/neutron/latest/admin/config-az.html

Reply:
@gouthampacha Im not saying programmatic way , but if operator specify in AllowedTopologies that means manila AZ is reachable from them I suppose. But if you see there are three options given by operator nova1, nova2, nova3 in the above example saying they are reachable to manila AZ , so you also want pod to be scheduled in either 3 of them right to be able to access storage if Im not wrong . If pod gets scheduled independent of these it might go into pending state . IIUC delayed binding takes care of that by taking into account the topology common to both. May be Im missing something in what you have mentioned.